### ISSUE TITLE
Incorrect Order of Layer Normalization in Transformer Block for Time Series Classification

### ISSUE THREAD

**Issue by Atousa-Kalantari** [keras-team/keras-io#1900]
### Issue Type

Support

### Source

source

### Keras Version

Keras 2.14

### Custom Code

Yes

### Current Behavior?

I noticed a discrepancy in the implementation of the Transformer model for time series classification in this repository. According to the original Transformer paper "Attention is All You Need", the residual connection should be added before applying the layer normalization. Specifically, the paper states:

"We employ a residual connection around each of the two sub-layers, followed by layer normalization. That is, the output of each sub-layer is LayerNorm(x + Sublayer(x)), where Sublayer(x) is the function implemented by the sub-layer itself."

However, in the current implementation, layer normalization is applied before the residual connection. Why?

### Standalone code to reproduce the issue or tutorial link

```python
Timeseries classification with a Transformer model

def transformer_encoder(inputs, head_size, num_heads, ff_dim, dropout=0):
    # Attention and Normalization
    x = layers.MultiHeadAttention(
        key_dim=head_size, num_heads=num_heads, dropout=dropout
    )(inputs, inputs)
    x = layers.Dropout(dropout)(x)
    x = layers.LayerNormalization(epsilon=1e-6)(x)
    res = x + inputs

    # Feed Forward Part
    x = layers.Conv1D(filters=ff_dim, kernel_size=1, activation="relu")(res)
    x = layers.Dropout(dropout)(x)
    x = layers.Conv1D(filters=inputs.shape[-1], kernel_size=1)(x)
    x = layers.LayerNormalization(epsilon=1e-6)(x)
    return x + res
```

**Comment by sachinprasadhs** [comment]
Hi, 

The implementation in https://keras.io/examples/timeseries/timeseries_classification_transformer is also correct, it is an improvement to the original paper by having Layer Normalization inside the block so that the training will be stable.

Here is a blog explaining the difference between both the approaches https://sh-tsang.medium.com/review-pre-ln-transformer-on-layer-normalization-in-the-transformer-architecture-b6c91a89e9ab
